src/Arduino_MKRIoTCarrier.cpp

```diff
@@ -31,22 +31,30 @@ int MKRIoTCarrier::_revision = 0;
 
 int MKRIoTCarrier::begin() {
 
-  pinMode(AREF_PIN,INPUT_PULLDOWN);

```

Comment:
AREF has an external pullup on rev2 while it's hanging on rev1, so the right thing to do is setting as INPUT_PULLDOWN and checking if it's LOW (rev1) or HIGH (rev2)

Reply:
the logic in last  version is inverted if it's LOW (rev2) or HIGH (rev1)